### ISSUE TITLE
GFANC for a headphone Pri and Sec Path.

### ISSUE THREAD

**Issue by JaySpeech** [Luo-Zhengding/Generative-Fixed-filter-Active-Noise-Control-based-on-Deep-Learning#2]
Thanks for your sharing for ANC. 
When I use a headphone Pri and Sec Path, I found GFANC can't work well.
In GFANC Paper, The Pri and Sec Path are bandpass FIR filter. I try to use our measure path, the mag response are below.

Pri Path:
![first_path_02](https://github.com/Luo-Zhengding/GFANC-Generative-fixed-filter-active-noise-control/assets/33861504/ddd13d4c-740f-4fc6-8448-fe50b1e4700c)

Sec Path:
![sec_path_02](https://github.com/Luo-Zhengding/GFANC-Generative-fixed-filter-active-noise-control/assets/33861504/51c294ae-6456-45b2-8634-2a8ab65a17d3)

I put 20~7980Hz white noise signal into train_fxlms_algorithm to create Pretrained_Sub_Control_filters. The fxlms output result is below.

![20_7980_fxlms](https://github.com/Luo-Zhengding/GFANC-Generative-fixed-filter-active-noise-control/assets/33861504/fa7f65a5-0b64-4710-8b0c-7c555de02549)

Using the above Pretrained_Sub_Control_filters to train. The Accuracy is not good. 

Epoch 37
Learning rate: 7.8125e-05
Training Loss: 0.12145824613980949 Training Accuracy: 0.5167499999999995
Validation Loss : 0.12397415339946746 Validation Accuracy : 0.5225000000000001

And the GFANC result for Aircraft nosie is also not good.

![GFANC1](https://github.com/Luo-Zhengding/GFANC-Generative-fixed-filter-active-noise-control/assets/33861504/efcbe366-5328-4952-bdd9-4dca0d3a980e)

I think there is a lot of nolinear in our pri and sec path and the fxlms can't get a ideal FIR filter.

**Comment by Luo-Zhengding** [comment]
Thanks for your question. If the actual paths in the scenario are not much different from the one we used for training, we can use our pre-trained sub control filters directly, but if the actual paths are much different from the one we used for training, you need to get the new sub control filters. It means that the trained neural network doesn't need to be changed, but you may need to retrain the broadband control filter and decompose it into sub control filters. Thus, only one pre-trained control filter is required as the prior data in the GFANC method.

**Comment by JaySpeech** [comment]
Thanks for your reply. I close the issue.
